path/basename.ts

```diff
@@ -6,20 +6,27 @@ import { basename as posixBasename } from "./posix/basename.ts";
 import { basename as windowsBasename } from "./windows/basename.ts";
 
 /**
- * Return the last portion of a `path`.
- * Trailing directory separators are ignored, and optional suffix is removed.
+ * Return the last portion of a path.
  *
- * @example
+ * The trailing directory separators are ignored, and optional suffix is
+ * removed.
+ *
+ * @example Usage
  * ```ts
  * import { basename } from "@std/path/basename";
+ * import { assertEquals } from "@std/assert/assert-equals";
  *
- * basename("/home/user/Documents/"); // "Documents"
- * basename("C:\\user\\Documents\\image.png"); // "image.png"
- * basename("/home/user/Documents/image.png", ".png"); // "image"
+ * if (Deno.build.os === "windows") {
+ *   assertEquals(basename("C:\\user\\Documents\\image.png"), "image.png");
+ * } else {
+ *   assertEquals(basename("/home/user/Documents/image.png"), "image.png");
+ * }
  * ```
```

Comment:
What if we wrote the OS-dependent functions like this? It clearly shows the differences in behaviors between difference OS systems.

Reply:
I'm not sure it's a good idea to use Deno specific API in examples as we specify this package as cross-runtime.

Reply:
Does `isWindows` evaluate to the correct OS on Node?

Reply:
Currently no. But I think we should add that support.

Reply:
I agree; we should add that support. But in the meantime, this package is only partially cross-platform and we should remove the other runtimes from the displayed list on JSR until it's truly cross-platform. I think it'd be good to have the assertions in these top-level functions, for now, until that's solved, as they guarantee correctness. My opinion isn't strong enough to block this PR. Up to you.

Reply:
See #4914

Reply:
Removed Node and Bun compatibility on JSR.

Added assertions to examples.